openstack/02a-MAClow/IEEE802154E.c

```diff
@@ -877,6 +893,27 @@ port_INLINE void activity_ti1ORri1() {
             }
             // record that I attempt to transmit this packet
             ieee154e_vars.dataToSend->l2_numTxAttempts++;
+            //START OF TELEMATICS CODE
+            /*
+             * if security is enabled on the current frame,
+             * I have to encrypt it before sending.
+             * Note that if this is not the first attempt, the
+             * clearText has to be encrypted again.
+             */
+            if(ieee154e_vars.dataToSend->l2_security == IEEE154_SEC_YES_SECURITY){
+            	//if it is a retransmission, recover the clearText packet
+            	if(ieee154e_vars.dataToSend->l2_numTxAttempts != 1){
+					 ieee154e_vars.dataToSend->length = ieee154e_vars.dataToSend->clearText_length;
+					 uint8_t i;
+					 for(i=0;i<ieee154e_vars.dataToSend->length; i++){
+						ieee154e_vars.dataToSend->l2_payload[i] = ieee154e_vars.dataToSend->clearText[i];
+					 }
+				 }
+
+            	security_outgoingFrame(ieee154e_vars.dataToSend);

```

Comment:
"secutiry" is a very general word. Would there be something more specific. `ccm_poipoi`?